linkerd/app/integration/src/client.rs

```diff
@@ -172,18 +183,39 @@ impl Client {
         self.running.await
     }
 
-    pub async fn shutdown(self) {
+    /// Shut down the client, returning a type that can be used to initiate a
+    /// new client connection to that target.
+    pub async fn shutdown(self) -> NewClient {
```

Comment:
You could even have this return `impl Fn() -> Client + Clone` -- not a lot of need to have a named type here...

Reply:
> You could even have this return `impl Fn() -> Client + Clone` -- not a lot of need to have a named type here...

hmm, I like the named type because it has a named _method_, the test is kinda unclear when you're not calling `.reconnect()` on something? but that's not a bad idea.